### DISCUSSION TITLE
Alpine x TailwindCSS: x-bind:disabled on array length not working

### DISCUSSION THREAD

**Discussion by sumanthratna** [alpinejs/alpine#665] [4. General]
screen recording: https://gofile.io/d/mrwhuX

```html
<div class="bg-teal-200 my-8 left-0 bottom-0 right-0 z-40 w-full shadow">
  <div class="container mx-auto px-4 py-4">
    <div class="flex md:items-center">
      <div class="mr-4 flex-shrink-0">
        <svg class="h-8 w-8 text-teal-600" viewBox="0 0 20 20" fill="currentColor"><path fill-rule="evenodd" d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z" clip-rule="evenodd" /></svg>
      </div>
      <div x-text="selectedRows.length + ' rows are selected'" class="text-teal-800 text-lg"></div>
      <button :disabled="Boolean(selectedRows.length===0)" class="ml-auto bg-gray-300 hover:bg-red-700 hover:text-white text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center disabled:opacity-50 disabled:cursor-not-allowed">DELETE</button>
      <button class="ml-6 bg-gray-300 hover:bg-gray-400 text-gray-800 font-bold py-2 px-4 rounded inline-flex items-center">
        <svg class="fill-current w-4 h-4 mr-2" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M13 8V2H7v6H2l8 8 8-8h-5zM0 18h20v2H0v-2z" /></svg>
        <span>DOWNLOAD CSV</span>
      </button>
    </div>
  </div>
</div>
```
As you can see in the screen recording, the disabled styles (`opacity-50` and `cursor-not-allowed`) aren't being applied when I select rows. However, when I select rows, the `disabled="disabled"` attribute is applied, so I know Alpine is doing *something*. I'm confused because even though Alpine has binded correctly, there's no visible change to the DOM. 

This happens in Version 84.0.4147.89, Firefox 78.0.2, and Safari 14.0 (16610.1.15.51.6) on macOS Big Sur. 

I don't think this is the cause of the problem, but this is in a Django project. Is this a me, Tailwind, or Alpine problem?

**Comment by SimoTod** [comment]
I believe it's just a tailwind misconfiguration. The disabled variant is not available by default: https://tailwindcss.com/docs/pseudo-class-variants/#default-variants-reference

**Reply by sumanthratna** [reply]
That seems reasonable; I'm not sure how I missed that :) However, when I set my `tailwind.config.js` to the following, I still have the same issue.
```js
module.exports = {
  purge: [],
  theme: {
    extend: {},
  },
  variants: {
    backgroundOpacity: ['responsive', 'hover', 'focus', 'disabled'],
    pointerEvents: ['responsive', 'disabled']
  },
  plugins: [],
}
```

**Reply by sumanthratna** [reply]
```js
module.exports = {
  purge: [],
  theme: {
    extend: {},
  },
  variants: {
    opacity: ['responsive', 'hover', 'focus', 'disabled'],
    backgroundOpacity: ['responsive', 'hover', 'focus', 'disabled'],
    pointerEvents: ['responsive', 'disabled']
  },
  plugins: [],
}
```
results in the opacity being changed. pointer-events-none still isn't working, but I think I can figure that out now. Thanks @SimoTod!
